### PR TITLE
Add parametric spring object

### PR DESCRIPTION
**Pull request by XPerianer**

### CODE REVIEW

File: src/database/node.rb [hunk 1]
```diff
@@ -41,11 +41,12 @@ def move_to(position)
...
   # Moves all connected components
   # this is very slow. Only do this if necessary (i.e. not in simulation)
+  # It is important to first update the hubs, as some edges will use the hub
   def update_sketchup_object
-    @incidents.each(&:update_sketchup_object)
-    @adjacent_triangles.each(&:update_sketchup_object)
     pods.each { |pod| pod.update_position(@position) }
     hub.entity.move!(Geom::Transformation.new(@position))
+    @incidents.each(&:update_sketchup_object)
+    @adjacent_triangles.each(&:update_sketchup_object)
```
Comment (birneamstiel) [inline src/database/node.rb]: Did you check that this doesn't break the conventional physics simulation?

File: src/models/parametric_spring_model.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
+
+class ParametricSpringModel
+  attr_reader :definition, :material
+
+  def initialize free_length, k
```
Comment (birneamstiel) [inline src/models/parametric_spring_model.rb]: I'm not sure about this k here... maybe we name the parameter radius for now?

File: src/models/parametric_spring_model.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    center_point = Geom::Point3d.new(1, 0, 0)
+    radius = 5 * k.mm
+
+    edgearray = entities.add_circle center_point, normal_vector, radius, 10
+    circle_face = entities.add_face(edgearray)
+    circle_face.followme curve
+    @material = Sketchup.active_model.materials['spring_material']
+  end
+
+  def generate_spring_curve(length, radius, windings, samples)
```
Comment (birneamstiel) [inline src/models/parametric_spring_model.rb]: Super nice way of visualizing the spring! 👍

File: src/database/node.rb (hunk 1 above)
Comment (XPerianer) [reply src/database/node.rb]: It does not seems so, although im not sure if it works in all potential cases

**Pull request merged**